libsoftwarecontainer/src/container.cpp

```diff
@@ -640,40 +644,73 @@ bool Container::bindMountCore(const std::string &pathInHost,
     std::string filePart = baseName(pathInContainer);
     std::string tempDirInContainer = buildPath(gatewaysDirInContainer(), filePart);
 
-    // The same paths on the host
-    std::string pathInContainerOnHost = buildPath(rootFS(), pathInContainer);
+    pid_t pid = INVALID_PID;
+    int status = -1;
+
+    // Utility lambda function to create parent paths of a given path
+    auto createParentDirectories = [this] (std::string path) {
+        std::stack<std::string> paths;
+        std::string currentPath = path;
+        while(!isDirectory(currentPath)) {
+            paths.push(currentPath);
+            currentPath = parentPath(currentPath);
+        }
+
+        while(!paths.empty()) {
+            std::string path = paths.top();
+            if (mkdir(path.c_str(), S_IRWXU | S_IRWXG | S_IRWXO) == -1) {
+                return 1;
+            }
+
+            paths.pop();
+        }
+        return 0;
+    };
 
     // Move the mount in the container if the tempdir is not the desired dir
     if (tempDirInContainer.compare(pathInContainer) != 0) {
         //
         // First, optionally create any parent directories to the targets
         // We do this on the host to avoid working too much inside the container
         //
+        std::string parentPathInContainer = parentPath(pathInContainer);
+        ExecFunction createParent = std::bind(createParentDirectories, parentPathInContainer);
+
+        if (!executeSync(createParent, &pid)) {
+            log_error() << "Could not create parent directory " << parentPathInContainer
+                        << " in container, status: " << status;
```

Comment:
Where the `status` get populated? Looks like you're going to return -1 always (line 648).